src/api/cpp/array.cpp

```diff
@@ -471,21 +471,31 @@ array::array_proxy &af::array::array_proxy::operator=(const array &other) {
         }
     }
 
-    af_array par_arr = nullptr;
+    af_array par_arr = 0;
 
+    dim4 dims = impl->parent_->dims();
     if (impl->is_linear_) {
         AF_THROW(af_flat(&par_arr, impl->parent_->get()));
+        // af_flat creates a new reference, remove the original reference
+        // in order to prevent triggering a memcpy in af_assign_gen when refCount > 1
```

Comment:
I think a better comment would be:

```
The set call will dereference the impl->parent_ array. We are doing
this because the af_flat call above increases the reference count of the
parent array which triggers a copy operation. This triggers a copy operation
inside the af_assign_gen function below. The parent array will be reverted
to the original array and shape later in the code.
```

Reply:
Done